dataactbroker/permissions.py

```diff
@@ -38,12 +42,20 @@ def decorated_function(*args, **kwargs):
                 elif LoginSession.isLogin(session):
                     user = sess.query(User).filter(User.user_id == session["name"]).one()
                     valid_user = True
-                    if permission in PERMISSION_TYPE_DICT and \
-                            not user.permission_type_id == PERMISSION_TYPE_DICT[permission]:
-                        valid_user = False
+
+                    if permission is not None and not user.permission_type_id == PERMISSION_TYPE_DICT['website_admin']:
+                        perm_hierarchy = {d['name']: d['order'] for d in PERMISSION_MAP.values()}
+                        # if the users permission is not higher than the one specified, check their permission
+                        # if user's perm order is < than what's passed in, it means they have higher permissions
+                        if perm_hierarchy[PERMISSION_TYPE_DICT_ID[user.permission_type_id]] > perm_hierarchy[permission]:
```

Comment:
Higher permissions have a lower `order` value so that when it's ordered it's a higher priority. that's why i had the comparison as it is above.